### DISCUSSION TITLE
Who maintains the data? Collaborate with DDF Guide?

### DISCUSSION THREAD

**Discussion by selmaohneh** [leifarriens/diedreifragezeichen-db#95] [Q&A]
Hi!

Who maintains the data inside the DB? Just you?

I have a ???-App called DDF Guide where I gather similar data, but in a plain .json file. 

https://github.com/selmaohneh/DdfGuide

You could use that file to add data like roles/speaker and authors per episode to the db. Maybe then I could migrate to your DB and link to your page...

Thoughts?

**Comment by leifarriens** [comment]
Hey @selmaohneh,
thank you for reaching out!

The data is maintained by an automated sync job, that checks the spotify API if a new episode is available. There is also a scraping job that gets the summary of the episodes, but I usually enter them manually.

If you would like to migrate to this database, there is an [API](https://github.com/leifarriens/diedreifragezeichen-db#api). You would need to create an API key, which I had to implement to prevent high usage costs on my side.

I also had a look at your project (♥️) und I see some potential incompatibilities:

- ddfdb has no data about the roles/speaker, which would need to be maintained manually. The original Idea of this project was, that it can "maintain itself", but maybe we can find a way to also automate the speakers.
- ddbdb does not have any data about "Die drei ??? kids" and "Die DR3I"

**Reply by selmaohneh** [reply]
* Data for "Kids" and "Die DR3I" should be available via the Spotify API, as well. I guess that could be added "easily".
* Are the scraping jobs part of this repo? Sorry... had no time to look yet. Maybe we could enhance those to also scrape author, speakers/roles und future episodes. That's a pain point for me, as well, which I wanted to automate for quite some time now.
* I'd be willing to support you with the usage costs if we'd achive to automate the whole data-collection process

**Comment by leifarriens** [comment]
> * Are the scraping jobs part of this repo? Sorry... had no time to look yet. Maybe we could enhance those to also scrape author, speakers/roles und future episodes. That's a pain point for me, as well, which I wanted to automate for quite some time now.

The scraping logic is defined in [this file](https://github.com/leifarriens/diedreifragezeichen-db/blob/production/src/services/inhalt.service.ts). I am using https://www.play-europa.de/ to get the data. The "Inhalte" are available on the top level, so I only need to do one network request per page (currently 14). 
I am currently changing the scraping job to point to https://www.dreifragezeichen.de/ since they stopped updating the data on play-europa.de (221 is the last entry).
To also get information about the "Sprecher" we will need to go into the details page of the episode and have one network request per folge.
Would it be enough for your app to have the Sprecher as one string?

> * Data for "Kids" and "Die DR3I" should be available via the Spotify API, as well. I guess that could be added "easily".

Sadly there is no official representation of "Die DR3I" folgen on any of the streaming platforms.
